polkadot/xcm/xcm-builder/src/location_conversion.rs

```diff
@@ -317,6 +317,27 @@ impl<Network: Get<Option<NetworkId>>, AccountId: From<[u8; 32]> + Into<[u8; 32]>
 	}
 }
 
+/// Extracts the `AccountId32` from the passed Treasury plurality if the network matches.
+pub struct TreasuryVoiceConvertsVia<Network, AccountId, TreasuryAccount>(
+	PhantomData<(Network, AccountId, TreasuryAccount)>,
+);
+impl<
+		Network: Get<Option<NetworkId>>,
+		AccountId: From<[u8; 32]> + Into<[u8; 32]> + Clone,
+		TreasuryAccount: Get<AccountId>,
+	> ConvertLocation<AccountId> for TreasuryVoiceConvertsVia<Network, AccountId, TreasuryAccount>
+{
+	fn convert_location(location: &MultiLocation) -> Option<AccountId> {
+		let id: [u8; 32] = match *location {
+			MultiLocation { parents: 0, interior: X1(Plurality { id: BodyId::Treasury, .. }) } |
+			MultiLocation { parents: 1, interior: X1(Plurality { id: BodyId::Treasury, .. }) } =>
+				TreasuryAccount::get().into(),
```

Comment:
This should be done via `HashedDescription` for `parents: 1`, can be kept local for `parents: 0`